File: src/main/java/com/raoulvdberge/refinedstorage/util/StorageBlockUtils.java
package com.raoulvdberge.refinedstorage.util;

import com.raoulvdberge.refinedstorage.RSTiles;
import com.raoulvdberge.refinedstorage.apiimpl.network.node.storage.StorageNetworkNode;
import com.raoulvdberge.refinedstorage.apiimpl.storage.ItemStorageType;
import com.raoulvdberge.refinedstorage.tile.StorageTile;
import net.minecraft.tileentity.TileEntityType;
import net.minecraft.util.ResourceLocation;

public class StorageBlockUtils {
    public static ResourceLocation getNetworkNodeId(ItemStorageType type) {
        switch (type) {
            case ONE_K:
                return StorageNetworkNode.ONE_K_STORAGE_BLOCK_ID;
            case FOUR_K:
                return StorageNetworkNode.FOUR_K_STORAGE_BLOCK_ID;
            case SIXTEEN_K:
                return StorageNetworkNode.SIXTEEN_K_STORAGE_BLOCK_ID;
            case SIXTY_FOUR_K:
                return StorageNetworkNode.SIXTY_FOUR_K_STORAGE_BLOCK_ID;
            case CREATIVE:
                return StorageNetworkNode.CREATIVE_STORAGE_BLOCK_ID;
            default:
                throw new IllegalArgumentException("Unknown storage type " + type);
        }
    }

    public static TileEntityType<StorageTile> getTileEntityType(ItemStorageType type) {
        switch (type) {
            case ONE_K:
                return RSTiles.ONE_K_STORAGE_BLOCK;
            case FOUR_K:
                return RSTiles.FOUR_K_STORAGE_BLOCK;
            case SIXTEEN_K:
                return RSTiles.SIXTEEN_K_STORAGE_BLOCK;
            case SIXTY_FOUR_K:
                return RSTiles.SIXTY_FOUR_K_STORAGE_BLOCK;
            case CREATIVE:
                return RSTiles.CREATIVE_STORAGE_BLOCK;
            default:
                throw new IllegalArgumentException("Unknown storage type " + type);
        }
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/util/StorageBlockUtils.kt
package com.refinedmods.refinedstorage.util

import com.refinedmods.refinedstorage.RSTiles
import com.refinedmods.refinedstorage.apiimpl.network.node.storage.StorageNetworkNode
import com.refinedmods.refinedstorage.apiimpl.storage.ItemStorageType
import com.refinedmods.refinedstorage.tile.StorageTile
import net.minecraft.tileentity.BlockEntityType
import net.minecraft.util.Identifier

object StorageBlockUtils {
    fun getNetworkNodeId(type: ItemStorageType): Identifier {
        return when (type) {
            ONE_K -> StorageNetworkNode.ONE_K_STORAGE_BLOCK_ID
            FOUR_K -> StorageNetworkNode.FOUR_K_STORAGE_BLOCK_ID
            SIXTEEN_K -> StorageNetworkNode.SIXTEEN_K_STORAGE_BLOCK_ID
            SIXTY_FOUR_K -> StorageNetworkNode.SIXTY_FOUR_K_STORAGE_BLOCK_ID
            CREATIVE -> StorageNetworkNode.CREATIVE_STORAGE_BLOCK_ID
            else -> throw IllegalArgumentException("Unknown storage type $type")
        }
    }

    fun getBlockEntityType(type: ItemStorageType): BlockEntityType<StorageTile> {
        return when (type) {
            ONE_K -> RSTiles.ONE_K_STORAGE_BLOCK
            FOUR_K -> RSTiles.FOUR_K_STORAGE_BLOCK
            SIXTEEN_K -> RSTiles.SIXTEEN_K_STORAGE_BLOCK
            SIXTY_FOUR_K -> RSTiles.SIXTY_FOUR_K_STORAGE_BLOCK
            CREATIVE -> RSTiles.CREATIVE_STORAGE_BLOCK
            else -> throw IllegalArgumentException("Unknown storage type $type")
        }
    }
}